### DISCUSSION TITLE
interval图，如何隐藏 tooltip

### DISCUSSION THREAD

**Discussion by wanglihuaya** [antvis/G2#5004] [Q&A] [answered]
interval图，如何隐藏 tooltip

chart.interaction('tooltip', {
  body:false,
});

复现代码
```javascript
import { Chart } from '@antv/g2';

const chart = new Chart({
  container: 'container',
  theme: 'classic',
  autoFit: true,
});

chart
  .interval()
  .data({
    type: 'fetch',
    value:
      'https://gw.alipayobjects.com/os/bmw-prod/fb9db6b7-23a5-4c23-bbef-c54a55fee580.csv',
  })
  .transform([{ type: 'sortX', by: 'y', reverse: true, slice: 5 }])
  .encode('x', 'letter')
  .encode('y', 'frequency')
  .axis('y', { labelFormatter: '.0%' });

  chart.interaction('tooltip', {
  body:false,
  crosshairs:false,
});

chart.render();

```

没有效果

**Comment by pearmini** [comment, marked as answer]
应该这样声明：`chart.interaction('tooltip', false)`

`chart.interaction('tooltip', { body:false});` 针对折线图，只展示指标线，不展示 body。

**Reply by hustcc** [reply]
还是以前的 api 好。

```ts
showCrosshair: boolean 
```

是否显示指标线，是比较适用的，为啥还有不显示 body。

**Reply by pearmini** [reply]
这个 api 现在也有：

```js
crosshair: boolean 
```

不显示 body 这个最开始只是给测试使用的，开发者一般不需要感知。

**Reply by wanglihuaya** [reply]
> 应该这样声明：`chart.interaction('tooltip', false)`
> 
> `chart.interaction('tooltip', { body:false});` 针对折线图，只展示指标线，不展示 body。

收到 🫡 over

**Reply by wanglihuaya** [reply]
> 这个 api 现在也有：
> 
> ```js
> crosshair: boolean 
> ```
> 
> 不显示 body 这个最开始只是给测试使用的，开发者一般不需要感知。

我看文档里 写的是body，没找到别的用法

**Reply by hustcc** [reply]
tooltip 在 ui 上分成两部分，分别是 body 和 crosshair，如下图所示：

![image](https://github.com/antvis/G2/assets/7856674/4ddf693a-ab1c-4147-9d4f-91bda081d8bf)

所以设置他们为 false，意思就是不显示这些 UI，如果要真正完全不使用 tooltip，就用以下代码完全关闭。

```ts
chart.interaction('tooltip', false)
```
